tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/crossprotocol/AMQPResenderToOpenwireTest.java

```diff
@@ -0,0 +1,158 @@
+      server.start();
+
+      SimpleString coreQueueZero = new SimpleString(queueZeroName);
+      server.createQueue(coreQueueZero, RoutingType.ANYCAST, coreQueueZero, null, false, false);
+
+      SimpleString coreQueueOne = new SimpleString(queueOneName);
+      server.createQueue(coreQueueOne, RoutingType.ANYCAST, coreQueueOne, null, false, false);
+   }
+
+   @Override
+   @After
+   public void tearDown() throws Exception {
+      if (server != null) {
+         server.stop();
+         server = null;
+      }
+   }
+
+   @Test(timeout = 60_000)
+   public void openwireInternalBinaryPropertiesShouldBeResentAsByteArrays() throws Exception {
+      final List<Throwable> errors = new LinkedList<>();
+      final ExecutorService executor = Executors.newFixedThreadPool(2);
+      final CountDownLatch resenderReady = new CountDownLatch(1);
+
+      // AMQP resender
+      executor.execute(new Runnable() {
+         @Override
+         public void run() {
+            try {
+
+               Connection connection = qpidFactory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+               Queue queueZero = session.createQueue(queueZeroName);
+               Queue queueOne = session.createQueue(queueOneName);
+
+               MessageConsumer consumer = session.createConsumer(queueZero);
+               connection.start();
+               resenderReady.countDown();
+               Message message = consumer.receive();
+               assertNotNull(message);
+
+               MessageProducer producer = session.createProducer(queueOne);
+               producer.send(message);
+
+               connection.close();
+
+            } catch (Exception e) {
+               errors.add(e);
+            }
+         }
+      });
+
+      // OpenWire client
+      executor.execute(new Runnable() {
+         @Override
+         public void run() {
+            try {
+
+               Connection connection = factory.createConnection();
+               Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+               Queue queueZero = session.createQueue(queueZeroName);
+               Queue queueOne = session.createQueue(queueOneName);
+
+               MessageProducer producer = session.createProducer(queueZero);
+               Message testMessage = session.createTextMessage("test");
+               resenderReady.await(20, TimeUnit.SECONDS);
+               producer.send(testMessage);
+
+               MessageConsumer consumer = session.createConsumer(queueOne);
+               connection.start();
+               Message receivedMessage = consumer.receive();
+               assertNotNull(receivedMessage);
+
+               connection.close();
+
+            } catch (Exception e) {
+               errors.add(e);
+            }
+         }
+      });
+
+      executor.shutdown();
+      executor.awaitTermination(20, TimeUnit.SECONDS);
```

Comment:
Can you reduce this timeout?

Reply:
I guess it can be reduced, will do after some testing